tests/lib/spread/rules/main.yaml

```diff
@@ -0,0 +1,23 @@
+rules:
+  tests:
+    from:
+      - tests/main/.*
+      - tests/core/.*
+      - tests/completion/.*
+      - tests/cross/.*
+      - tests/regression/.*
+      - tests/smoke/.
+      - tests/unit/.*
+      - tests/upgrade/.*
+    to: [$SELF]
```

Comment:
so if we change a test we just run that test? what if it's restore is wrong and it will affect other tests?

Reply:
Right, if the change breaks the restore this could affect following tests

As an alternative we could run:
to: [ $SELF, /path/to/sanity/suite ]

So we make sure we always cover smoke or a sanity suite with any change

Reply:
I propose that because, we have code reviews, and also the prepare-restore helpers make sure the environment is cleaned after each test is executed. So, installed snaps/debs are deleted, test directory is cleaned, it is verified that a big part of the file system has not changed, etc

Reply:
you probably have more a sense than me how oftern we get test interference problems? anyway on master we will run all the tests?

Reply:
yes, finally if the there is no pr, we will run everything.